catroid/src/org/catrobat/catroid/web/ServerCalls.java

```diff
@@ -344,6 +344,12 @@ public boolean registerOrCheckToken(String username, String password, String use
 			userEmail = emailForUiTests;
 		}
 
+		if (userEmail == null) { //restricted users
```

Comment:
If you have to write a comment which exactly describes the condition, create a method instead.